File: update_tracker.js
const fs = require('fs');
let code = fs.readFileSync('frontend/src/app/page.tsx', 'utf8');

const stateCodeToAdd = `  const [completedChapters, setCompletedChapters] = useState<string[]>([]);
  
  // Tracker State
  const [expandedTestaments, setExpandedTestaments] = useState<string[]>([]);
  const [expandedBooks, setExpandedBooks] = useState<string[]>([]);
  
  const toggleTestament = (t: string) => setExpandedTestaments(prev => prev.includes(t) ? prev.filter(x => x !== t) : [...prev, t]);
  const toggleBook = (b: string) => setExpandedBooks(prev => prev.includes(b) ? prev.filter(x => x !== b) : [...prev, b]);
`;
code = code.replace(`  const [completedChapters, setCompletedChapters] = useState<string[]>([]);`, stateCodeToAdd);

const oldTrackerUI = `                <div className="text-[13px] font-bold tracking-widest text-[#87867f] uppercase mb-6 ml-2">Old Testament</div>
                <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-6 mb-16">
                  {OT_BOOKS.map(book => (
                    <div key={book.name} className="bg-[#30302e] p-6 rounded-[20px] ring-shadow">
                      <h3 className="font-medium text-[16px] text-[#faf9f5] mb-5">{book.name}</h3>
                      <div className="flex flex-wrap gap-2">
                        {Array.from({ length: book.chapters }).map((_, i) => {
                          const id = \`\${book.name}-\${i + 1}\`;
                          const isChecked = completedChapters.includes(id);
                          return (
                            <button
                              key={i}
                              onClick={() => toggleAnyChapter(id)}
                              className={\`w-11 h-11 lg:w-9 lg:h-9 rounded-xl text-xs font-semibold flex items-center justify-center transition-all \${
                                isChecked 
                                  ? 'bg-[#c96442] text-white shadow-sm' 
                                  : 'bg-[#141413] text-[#87867f] hover:text-[#faf9f5] hover:bg-[#4d4c48]'
                              }\`}
                            >
                              {i + 1}
                            </button>
                          )
                        })}
                      </div>
                    </div>
                  ))}
                </div>

                <div className="text-[13px] font-bold tracking-widest text-[#87867f] uppercase mb-6 ml-2">New Testament</div>
                <div className="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-6 mb-16">
                  {NT_BOOKS.map(book => (
                    <div key={book.name} className="bg-[#30302e] p-6 rounded-[20px] ring-shadow">
                      <h3 className="font-medium text-[16px] text-[#faf9f5] mb-5">{book.name}</h3>
                      <div className="flex flex-wrap gap-2">
                        {Array.from({ length: book.chapters }).map((_, i) => {
                          const id = \`\${book.name}-\${i + 1}\`;
                          const isChecked = completedChapters.includes(id);
                          return (
                            <button
                              key={i}
                              onClick={() => toggleAnyChapter(id)}
                              className={\`w-11 h-11 lg:w-9 lg:h-9 rounded-xl text-xs font-semibold flex items-center justify-center transition-all \${
                                isChecked 
                                  ? 'bg-[#c96442] text-white shadow-sm' 
                                  : 'bg-[#141413] text-[#87867f] hover:text-[#faf9f5] hover:bg-[#4d4c48]'
                              }\`}
                            >
                              {i + 1}
                            </button>
                          )
                        })}
                      </div>
                    </div>
                  ))}
                </div>`;

const newTrackerUI = `                {/* Old Testament Accordion */}
                <div className="mb-6">
                  <button 
                    onClick={() => toggleTestament('OT')}
                    className="w-full flex items-center justify-between text-left bg-[#30302e] p-5 rounded-[20px] ring-shadow hover:bg-[#393936] transition-colors"
                  >
                    <div className="text-[16px] font-bold tracking-widest text-[#faf9f5] uppercase">Old Testament</div>
                    <div className="text-[#87867f]">{expandedTestaments.includes('OT') ? '▲' : '▼'}</div>
                  </button>
                  
                  {expandedTestaments.includes('OT') && (
                    <div className="mt-4 flex flex-col gap-3 pl-4 border-l-2 border-[#30302e]">
                      {OT_BOOKS.map(book => {
                        const isBookExpanded = expandedBooks.includes(book.name);
                        return (
                          <div key={book.name} className="bg-[#1c1c1b] rounded-[16px] overflow-hidden ring-1 ring-[#30302e]">
                            <button 
                              onClick={() => toggleBook(book.name)}
                              className="w-full flex items-center justify-between p-4 hover:bg-[#252523] transition-colors"
                            >
                              <h3 className="font-medium text-[15px] text-[#faf9f5]">{book.name}</h3>
                              <span className="text-xs text-[#87867f] font-mono">{book.chapters} CH</span>
                            </button>
                            
                            {isBookExpanded && (
                              <div className="p-4 pt-0 border-t border-[#30302e] bg-[#141413]">
                                <div className="flex flex-wrap gap-2 mt-4">
                                  {Array.from({ length: book.chapters }).map((_, i) => {
                                    const id = \`\${book.name}-\${i + 1}\`;
                                    const isChecked = completedChapters.includes(id);
                                    return (
                                      <button
                                        key={i}
                                        onClick={() => toggleAnyChapter(id)}
                                        className={\`w-11 h-11 lg:w-9 lg:h-9 rounded-xl text-xs font-semibold flex items-center justify-center transition-all \${
                                          isChecked 
                                            ? 'bg-[#c96442] text-white shadow-sm' 
                                            : 'bg-[#252523] text-[#87867f] hover:text-[#faf9f5] hover:bg-[#4d4c48]'
                                        }\`}
                                      >
                                        {i + 1}
                                      </button>
                                    )
                                  })}
                                </div>
                              </div>
                            )}
                          </div>
                        );
                      })}
                    </div>
                  )}
                </div>

                {/* New Testament Accordion */}
                <div className="mb-16">
                  <button 
                    onClick={() => toggleTestament('NT')}
                    className="w-full flex items-center justify-between text-left bg-[#30302e] p-5 rounded-[20px] ring-shadow hover:bg-[#393936] transition-colors"
                  >
                    <div className="text-[16px] font-bold tracking-widest text-[#faf9f5] uppercase">New Testament</div>
                    <div className="text-[#87867f]">{expandedTestaments.includes('NT') ? '▲' : '▼'}</div>
                  </button>
                  
                  {expandedTestaments.includes('NT') && (
                    <div className="mt-4 flex flex-col gap-3 pl-4 border-l-2 border-[#30302e]">
                      {NT_BOOKS.map(book => {
                        const isBookExpanded = expandedBooks.includes(book.name);
                        return (
                          <div key={book.name} className="bg-[#1c1c1b] rounded-[16px] overflow-hidden ring-1 ring-[#30302e]">
                            <button 
                              onClick={() => toggleBook(book.name)}
                              className="w-full flex items-center justify-between p-4 hover:bg-[#252523] transition-colors"
                            >
                              <h3 className="font-medium text-[15px] text-[#faf9f5]">{book.name}</h3>
                              <span className="text-xs text-[#87867f] font-mono">{book.chapters} CH</span>
                            </button>
                            
                            {isBookExpanded && (
                              <div className="p-4 pt-0 border-t border-[#30302e] bg-[#141413]">
                                <div className="flex flex-wrap gap-2 mt-4">
                                  {Array.from({ length: book.chapters }).map((_, i) => {
                                    const id = \`\${book.name}-\${i + 1}\`;
                                    const isChecked = completedChapters.includes(id);
                                    return (
                                      <button
                                        key={i}
                                        onClick={() => toggleAnyChapter(id)}
                                        className={\`w-11 h-11 lg:w-9 lg:h-9 rounded-xl text-xs font-semibold flex items-center justify-center transition-all \${
                                          isChecked 
                                            ? 'bg-[#c96442] text-white shadow-sm' 
                                            : 'bg-[#252523] text-[#87867f] hover:text-[#faf9f5] hover:bg-[#4d4c48]'
                                        }\`}
                                      >
                                        {i + 1}
                                      </button>
                                    )
                                  })}
                                </div>
                              </div>
                            )}
                          </div>
                        );
                      })}
                    </div>
                  )}
                </div>`;

if (code.includes(oldTrackerUI)) {
  code = code.replace(oldTrackerUI, newTrackerUI);
  fs.writeFileSync('frontend/src/app/page.tsx', code);
  console.log("Success tracker");
} else {
  console.log("Could not find old tracker UI");
}

File: frontend/update_tracker.py
import re

with open('src/app/(main)/page.tsx', 'r') as f:
    content = f.read()

# 1. Add settings state
state_block = r"(const \[activeNoteId, setActiveNoteId\] = useState<string \| null>\(null\);)"
new_state = r"""\1
  const [isSettingsOpen, setIsSettingsOpen] = useState(false);
  const [theme, setTheme] = useState('dark');
  const [trackerFormat, setTrackerFormat] = useState<'percent' | 'fraction'>('percent');

  useEffect(() => {
    const savedTheme = localStorage.getItem('theme') || (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light');
    setTheme(savedTheme);
    const savedTracker = localStorage.getItem('trackerFormat') as 'percent' | 'fraction' || 'percent';
    setTrackerFormat(savedTracker);
  }, []);

  const toggleTheme = () => {
    const newTheme = theme === 'dark' ? 'light' : 'dark';
    setTheme(newTheme);
    localStorage.setItem('theme', newTheme);
    if (newTheme === 'light') document.documentElement.setAttribute('data-theme', 'light');
    else document.documentElement.removeAttribute('data-theme');
  };

  const toggleTrackerFormat = () => {
    const newFormat = trackerFormat === 'percent' ? 'fraction' : 'percent';
    setTrackerFormat(newFormat);
    localStorage.setItem('trackerFormat', newFormat);
  };
"""
content = re.sub(state_block, new_state, content)

# 2. Add Settings Modal UI inside the return block, before {/* MOBILE NAV */}
modal_anchor = r"(\{/\* MOBILE NAV \*/\})"
settings_modal = r"""
        {/* SETTINGS MODAL */}
        {isSettingsOpen && (
          <div className="fixed inset-0 bg-black/60 backdrop-blur-sm z-[100] flex items-center justify-center p-4">
            <div className="bg-bg w-full max-w-sm rounded-[24px] p-6 shadow-2xl ring-1 ring-border">
              <div className="flex items-center justify-between mb-6">
                <h2 className="text-xl font-display text-fg">Settings</h2>
                <button onClick={() => setIsSettingsOpen(false)} className="text-muted hover:text-fg transition-colors">✕</button>
              </div>
              
              <div className="flex flex-col gap-4">
                <div className="flex items-center justify-between p-4 bg-surface rounded-[16px] ring-1 ring-border">
                  <div>
                    <div className="text-[15px] font-medium text-fg">Appearance</div>
                    <div className="text-[13px] text-muted">{theme === 'dark' ? 'Dark Mode' : 'Light Mode'}</div>
                  </div>
                  <button onClick={toggleTheme} className="w-12 h-6 rounded-full bg-border-soft relative transition-colors" style={{ backgroundColor: theme === 'light' ? 'var(--accent)' : 'var(--border-soft)' }}>
                    <div className="w-5 h-5 rounded-full bg-fg absolute top-0.5 transition-transform" style={{ transform: theme === 'light' ? 'translateX(26px)' : 'translateX(2px)' }} />
                  </button>
                </div>
                
                <div className="flex items-center justify-between p-4 bg-surface rounded-[16px] ring-1 ring-border">
                  <div>
                    <div className="text-[15px] font-medium text-fg">Tracker Format</div>
                    <div className="text-[13px] text-muted">{trackerFormat === 'percent' ? 'Percentage (%)' : 'Fractions (1/10)'}</div>
                  </div>
                  <button onClick={toggleTrackerFormat} className="text-[13px] font-semibold bg-bg px-3 py-1.5 rounded-lg text-fg ring-1 ring-border hover:bg-surface-warm transition-colors">
                    Toggle
                  </button>
                </div>
              </div>
            </div>
          </div>
        )}
        
        \1"""
content = re.sub(modal_anchor, settings_modal, content)

# 3. Add Settings button to Sidebar
sidebar_anchor = r"(\{/\* Sidebar User/Signout \*/\})"
settings_btn = r"""
            <button onClick={() => setIsSettingsOpen(true)} className="flex items-center gap-3 px-4 py-3 text-[#87867f] hover:text-[#faf9f5] hover:bg-[#30302e] rounded-xl transition-all font-medium text-[15px] group w-full text-left">
              <Settings size={18} className="group-hover:rotate-45 transition-transform duration-300" />
              Settings
            </button>
            \1"""
content = re.sub(sidebar_anchor, settings_btn, content)

# Also import Settings icon from lucide-react
import_lucide = r"(import \{.*?)(\} from 'lucide-react';)"
content = re.sub(import_lucide, r"\1, Settings \2", content)

with open('src/app/(main)/page.tsx', 'w') as f:
    f.write(content)

print("Settings added!")
